File: Models/Case.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MtpApp.Models
{
    public class Case
    {
        public int Id { get; set; }

        [Required]
        public int ClientId { get; set; }

        public Client Client { get; set; }

        [Required]
        public int CaseWorkerId { get; set; }

        public CaseWorker CaseWorker { get; set; }

        [Required]
        [StringLength(50)]
        public string Priority { get; set; }

        [Required]
        [StringLength(50)]
        public string ServiceType { get; set; }

        [StringLength(255)]
        public string Subject { get; set; }

        [StringLength(510)]
        public string Description { get; set; }

        [Required]
        public DateTime OpenDate { get; set; }

        public DateTime? CloseDate { get; set; }

        [StringLength(50)]
        public string Status { get; set; }
    }
}

File: spring-backend/src/main/java/com/mtp/api/models/Case.java
package com.mtp.api.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "Cases")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Case {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ClientId")
    private Client client;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "CaseWorkerId")
    private CaseWorker caseWorker;

    @NotNull
    @Size(max = 50)
    private String priority;

    @NotNull
    @Size(max = 50)
    private String serviceType;

    @Size(max = 255)
    private String subject;

    @Size(max = 510)
    private String description;

    @NotNull
    private LocalDateTime openDate;

    private LocalDateTime closeDate;

    @Size(max = 50)
    private String status;
}
